bot/seasons/pride/pride_facts.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+import json
+import logging
+import random
+from datetime import datetime
+from pathlib import Path
+from typing import Union
+
+import dateutil.parser
+import discord
+from discord.ext import commands
+
+from bot.constants import Channels
+from bot.constants import Colours
+
+log = logging.getLogger(__name__)
+
+Sendable = Union[commands.Context, discord.TextChannel]
+
+
+class PrideFacts(commands.Cog):
+    """Provides a new fact every day during the Pride season!"""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.facts = self.load_facts()
+
+    @staticmethod
+    def load_facts() -> dict:
+        """Loads a dictionary of years mapping to lists of facts."""
+        with open(Path("bot/resources/pride/facts.json"), "r", encoding="utf-8") as f:
+            return json.load(f)
+
+    async def send_pride_fact_daily(self) -> None:
+        """Background task to post the daily pride fact every day."""
+        channel = self.bot.get_channel(Channels.seasonalbot_chat)
+        while True:
+            await self.send_select_fact(channel, datetime.utcnow())
+            await asyncio.sleep(24 * 60 * 60)
+
+    async def send_random_fact(self, ctx: commands.Context) -> None:
```

Comment:
I think it would be better for random facts to use all of the facts as the pool from which to draw from rather than limiting it to the previous + current years.

Reply:
> I think it would be better for random facts to use all of the facts as the pool from which to draw from rather than limiting it to the previous + current years.

That would go against what was requested in the original issue:
> However, make sure that it doesn't expose future facts of the day, just to keep things a bit surprising.

@SebastiaanZ Thoughts?

I don't have strong feelings one way or the other.

Reply:
That's fine then, I hadn't read the issue itself.